Added recommendation about module name (#49540)
ansible/ansible: docs/docsite/rst/dev_guide/developing_modules_best_practices.rst

=========================

* Each module should be self-contained in one file, so it can be be auto-transferred by Ansible.

> * Module name should be using underscores instead of hyphens or spaces as word separator. Using hyphens and spaces will prevent Ansible from importing your module.

Reviewer: ```suggestion
* Module name MUST use underscores instead of hyphens or spaces as word separator. Using hyphens and spaces will prevent Ansible from importing your module.
```

* Do NOT use wildcards (*) for importing other python modules; instead, list the function(s) you are importing (for example, ``from some.other_python_module.basic import otherFunction``).
* Import custom packages in ``try``/``except``, capture any import errors, and handle them with ``fail_json()`` in ``main()``. For example:

.. code-block:: python

    import traceback

    from ansible.basic import missing_required_lib

    LIB_IMP_ERR = None

>     try:

Author: This part was with mixed tabs and spaces.

        import foo
        HAS_LIB = True
    except:
        HAS_LIB = False
        LIB_IMP_ERR = traceback.format_exc()

Then in ``main()``, just after the argspec, do

.. code-block:: python

    if not HAS_LIB:

>         module.fail_json(msg=missing_required_lib("foo"),

Author: same here.
